docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,112 @@
+
+    - Refactor relevant overlay methods to handle downtime_suspended nodes: `KnownReliable`, `FindNodes`, etc.
+    - Add notification to SN dashboard indicating suspension for downtime.
+
+4. Implement email notifications if node becomes suspended
+
+5. Implement evaluation chore to reinstate and disqualify nodes
+    - Check total downtime for each suspended node where last_contact_success > last_contact_failure. If the total downtime has fallen below the allowed downtime, the node's suspension is lifted, though it still remains under review.
+    - If node.under_review <= now - (grace period + tracking period) it is eligible for evaluation
+    - If the node is eligible for evaluation, we must also ensure that the entire tracking period is accounted for. If last contact was a failure and occurred before the end of the tracking period, we might have a window of downtime which has not yet been recorded. Thus, to measure all downtime for the tracking period, we must also wait until last_contact_success > failure OR last_contact_failure >= end of tracking period.
+
+        When these conditions are met, measurements should take care to include all downtime within the period. In addition to the lower bound issue explained above under step 3, we also have an upper bound issue in this case. We may have an entry which exists beyond the upper bound of the tracking period, yet holds some amount of downtime which occured within it.<br>
+        <br>
+        For example, if we want to measure the total downtime for node A from March 1st 00:00 to March 31st 00:00, and we have an entry from April 1st 01:00 indicating 2 hours of downtime, we must include the one hour of downtime which occurred within the tracking period in our calculation. 
+
+## Wrapup
+
+- The person that implements step 5 above should archive this document.
+
+- The [Disqualification blueprint](disqualification.md), and possibly the whitepaper, will need to be updated to reflect new up/downtime disqualification mechanic.
+
+- Edit [Audit Suspension blueprint](audit-suspend.md) to reflect change from `suspended` to `audit_suspended`
+
+- Link to this document in [Downtime Tracking](storage-node-downtime-tracking.md)
+
+## Open issues
+
+- There is a way to get around downtime tracking in its current implementation. Only come online long enough to check in then immediately shut down. Repeat this process frequently enough and a node can effectively remain offline and never be punished for it. This is due to the way downtime tracking selects nodes where last contact success < last contact failure. By checking in frequently enough, a node can reduce the risk that its last contact was a failure when the estimation chore runs.
```

Comment:
I'd say this is significantly worse than its portrayed here... The current downtime tracking service doesn't require the SN to do any meaningful work; it just proves the user controls an IP.  I could write a "ping responder" that always passes uptime checks.  I'll never fail an audit because I never accept data.  I'll always be a "new node" and strangle legit new nodes out of the network.

Reply:
I think we need to go back to audit-only uptime checking and we also need something to DQ nodes which never do any work at all (never have anything auditable).

Reply:
This also worries me the most. We're going from something that measures if the node works ("can you respond to audits?") to something that can be much more easily gamed ("can you issue checkins with a satellite on a regular interval?").

In addtion to that not measuring the actual property we require to have a healthy network, it's no longer randomized, which means that nodes can invent strategies to defeat it with certainty, rather than with low expectation. It worries me that we'll constantly be fighting an uphill battle against smarter attacks.

Previously, under correct operation, a node with more data, and hence more impact on the network, would get chosen for audits more often. That's a desirable property. If you're more likely to cause damage to the network if you go offline, you should have more incentive to stay online. I see no such consideration given with this design.

I think I would much prefer something closer to the weighted changes to the beta-distribution parameters, or something similar. There seems to be a ton of design space not discussed in here to fix the specific problem of "too many audits came in for a node while it was offline for a short time."

Reply:
How does "ping responder" node strange legit new nodes out of the network?

Reply:
Yeah, you're right. There are probably some potential solutions here that haven't been given enough thought. I'll see if I can add some new ideas. Thanks for the feedback

Reply:
@zeebo I have some ideas of how we might be able to modify things to resolve your concerns while still having more accurate downtime metrics

First, if we make sure we are completely exhausting the audit queue before replacing it with a new one, that makes everything else a lot easier - it means that each node on our network got at least three randomized pings over the course of the audit queue duration (which we would need to determine somehow).

Second, as well as updating node.LastContactSuccess and node.LastContactFailure, we can store data into our downtime tracking database: timestamp, node ID, ping result. Node ID is probably the largest data type here, so we should think of ways to reduce that so we can have lots of entries.

Now, for each node, we can iterate over each entry in the downtime tracking table to calculate an estimated amount of time offline over any arbitrary interval. That way pings closer together are not treated very differently than the pings further apart. The only issue is that we don't have a good way of resolving buffer issues (including the time between a LCS and LCF or vice-versa). But if we can be sure nodes are being pinged enough times, the time should be pretty accurate.

Another issue might be database size and performance, but I am not sure what we can do about that. We're already updating overlay cache for every node for every audit, so what's another table?

Reply:
@VinozzZ I think what he's getting at is if there were enough of these entities, the pool of new nodes could be diluted and legitimate new nodes would be less likely to be picked for uploads

Reply:
^^^ yes

Reply:
I'm going to be adding more ideas to this document, and I do agree with the sentiment that audits are the best indicator that a node is up and running _properly_, but I also wanted to mention a potential solution to this problem littleskunk came up with - a new kind of containment mode.

A node is audited, found to be offline, but it was last seen to be online- it enters containment mode.
It's audited again, still offline, no penalty, though it still remains in containment.
It's audited again, offline, but again last contact was found to be successful, increment containment counter. After some number of iterations it counts as a failed audit.

Reply:
Let's meet to talk about these new concerns and try to come out of the meeting with a proposal